adminguide.rst

```diff
@@ -0,0 +1,514 @@
+
+ $ sudo -u flux flux keygen
+ Saving /home/flux/.flux/curve/client
+ Saving /home/flux/.flux/curve/client_private
+ Saving /home/flux/.flux/curve/server
+ Saving /home/flux/.flux/curve/server_private
+
+
+.. _configuration:
+
+-----------------------------
+System Instance Configuration
+-----------------------------
+
+Much of Flux configuration occurs via
+`TOML <https://github.com/toml-lang/toml>`_ configuration files found
+in a hierarchy under ``/etc/flux``.  For the most part, Flux
+components will read all TOML files from a given directory in a glob,
+e.g. ``/etc/flux/component/*.toml``, which allows TOML tables to be
+combined in a single file or split across multiple files. For example,
+a typical Flux system instance will read all configuration from
+``/etc/flux/system/conf.d/*.toml``.
+
+In this guide, separate files will typically be used for clarity, instead
+of adding all configuration tables to a single TOML file.
+
+
+.. _configuration-security:
+
+^^^^^^^^^^^^^^^^^^^^
+flux-security config
+^^^^^^^^^^^^^^^^^^^^
+
+In order to run multi-user workloads ``flux-security`` components such
+as the signing library and ``flux-imp`` need proper configuration.
+
+First, a valid signing method should be configured. In
+``/etc/flux/security/conf.d/sign.toml``, add the following to configure
+job requests be signed using MUNGE:
+
+.. code-block:: toml
+
+ [sign]
+ max-ttl = 1209600  # 2 weeks
+ default-type = "munge"
+ allowed-types = [ "munge" ]
+
+
+Then configure ``flux-imp`` by creating ``/etc/flux/imp/exec.toml``
+with the following contents:
+
+.. code-block:: toml
+
+ [exec]
+ allowed-users = [ "flux" ]
+ allowed-shells = [ "/usr/libexec/flux/flux-shell" ]
+
+
+This ensures that only the ``flux`` user may run the ``flux-imp`` executable,
+and the only allowed job shell is the system installed ``flux-shell``.
+
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Execution system configuration
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+A system Flux instance must be configured to use a ``flux-imp`` process
+as a privileged helper for multi-user execution. This configuration should
+be made in ``/etc/flux/system/conf.d/exec/toml``. This configuration table
+is read by the ``job-exec`` module.
+
+.. code-block:: toml
+
+ [exec]
+ imp = "/usr/libexec/flux/flux-imp"
+
+
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Instance access configuration
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+By default, a Flux instance does not allow access to any user other than
+the instance *owner* (the user running the flux brokers). This is not
+suitable for a system instance, so *guest user* access should be enabled
+in ``/etc/flux/system/conf.d/access.toml``:
+
+.. code-block:: toml
+
+ [access]
+ allow-guest-user = true
+
+For a system instance it may be convenient to allow the ``root`` user to
+act as the instance owner (so that ``root`` can list and cancel jobs,
+access the KVS, etc). If this is desired, then ``allow-root-owner = true``
+should be set in the ``[access]`` table as well.
```

Comment:
Could mention that the alternative is to require administrators to run Flux administration commands as the flux user?